core/src/main/scala/kafka/utils/Json.scala

```diff
@@ -35,16 +35,7 @@ object Json {
    */
   def parseFull(input: String): Option[JsonValue] =
```

Comment:
I'm probably missing something obvious, but I found the following callers of this function: `DeleteRecordsCommand`, `LeaderElectionCommand`, `PreferredReplicaLeaderElectionCommand`, and `ReassignPartitionsCommand`. As far as I can tell, none of these uses involve the parsing of ACLs. Did we lose this compatibility handling at some point or is there some magic invocation?

Reply:
The compatibility fix was only required for ACLs, but we added it here to keep the fix small for backporting. The idea was to move to the appropriate place soon after, but it got stuck for a while. Do you have some concerns?

Reply:
Discussed this offline with @hachikuji. I had misunderstood his point. It looks like the compatibility code has not been used since 1.1.0:

https://github.com/apache/kafka/commit/9b44c3e7f89b55528b12ba280007c804682beb0b#diff-81f2ec590f4e047be7e3a7e5267cbc9aR60

Reply:
Maybe we should just drop this code.

Reply:
I looks like `parseFull` is used in a couple of places but I can perhaps just delegate this to `parseBytes` like this:
```
def parseFull(input: String): Option[JsonValue] = parseBytes(input.getBytes(Charset.defaultCharset()))
```
Or shall I just get rid of this code and use `parseBytes` everywhere?

Reply:
This code is fine as it is. We are suggesting that we don't need `parseBytesWithAclFallback` in `AclEntry`.

Reply:
Ok, then I misunderstood it too. :)
However isn't it possible that this issue resurfaces again if we remove this? I suspect the user who raised it may still use 1.0 where it got fixed and when they upgrade they'd run into this again. Or am I missing something else too?

Reply:
@viktorsomogyi I think the argument is that if no-one complained since 1.1 which was released more than 2 years ago, then maybe this issue is very rare.

Reply:
Ok, that's fair. I've uploaded the changes.